packages/ibm-products/src/components/ConditionBuilder/ConditionGroupBuilder/ConditionGroupBuilder.js

```diff
@@ -33,29 +35,52 @@ const ConditionGroupBuilder = ({
     }
   };
 
-  const onChangeHandler = (updatedConditions, conditionIndex) => {
+  const onChangeHandler = (updatedCondition, conditionId) => {
     onChange({
-      ...state,
-      conditions: state.conditions.map((condition, cIndex) =>
-        conditionIndex === cIndex ? updatedConditions : condition
+      ...group,
+      conditions: group.conditions.map((condition) =>
```

Comment:
similar comment about updating the array without the usage of `map`

Reply:
modified by using slice()